src/preflight/links-checks.js

```diff
@@ -41,45 +43,85 @@ export async function runInternalLinkChecks(urls, scrapedObjects, context, optio
         const anchors = Array.from(doc.querySelectorAll('a[href]'));
         const pageOrigin = new URL(pageUrl).origin;
         const internalSet = new Set();
+        const externalSet = new Set();
 
         anchors.forEach((a) => {
           try {
             const abs = new URL(a.href, pageUrl).toString();
             if (new URL(abs).origin === pageOrigin) {
               internalSet.add(abs);
+            } else {
+              externalSet.add(abs);
             }
           } catch {
             // skip invalid hrefs
           }
         });
 
         log.info('[preflight-audit] Found internal links:', internalSet);
+        log.info('[preflight-audit] Found external links:', externalSet);
 
+        // Check internal links
         await Promise.all(
           Array.from(internalSet).map(async (href) => {
             try {
-              const res = await fetch(href, {
+              const response = await fetch(href, {
                 method: 'HEAD',
                 headers: {
                   Authorization: options.pageAuthToken,
                 },
+                timeout: 3000,
               });
-              if (res.status >= 400) {
-                log.debug(`[preflight-audit] url ${href} returned with status code: %s`, res.status, res.statusText);
-                brokenInternalLinks.push({ urlTo: href, href: pageUrl, status: res.status });
+              const { status } = response;
+
+              if (status >= 400) {
+                log.debug(`[preflight-audit] internal url ${href} returned with status code: %s`, status);
+                brokenInternalLinks.push({ urlTo: href, href: pageUrl, status });
               }
             } catch (err) {
               log.error(`[preflight-audit] Error checking internal link ${href} from ${pageUrl}:`, err.message);
             }
           }),
         );
+
+        // Check external links
+        await Promise.all(
+          Array.from(externalSet).map(async (href) => {
+            try {
+              const response = await fetch(href, {
+                method: 'HEAD',
+                headers: {
+                  Authorization: options.pageAuthToken,
```

Comment:
@tathagat2241 , We shouldn’t use the access token with external links - it’s not required to access them and poses a potential security risk, as the token might be logged, retrieved, and used by third parties.